### ISSUE TITLE
IgnoreOperationAnnotation handling is too broken

### ISSUE THREAD

**Issue by ialidzhikov** [gardener/gardener#3321]
/kind bug
/priority normal

**What happened**:
https://github.com/gardener/gardener/pull/2290 introduces a bunch of `IgnoreOperationAnnotation`s in the `AddArgs` of the controllers but currently no one of the extension controller is passing this `IgnoreOperationAnnotation` and it is always evaluated to false. Other than that for the backupentry and the backupbucket controllers the reconcileOpts are not applied to the corresponding `DefaultAddOptions`.
Currently we have if statements that check `IgnoreOperationAnnotation` in `Add(mgr manager.Manager, args AddArgs)` funcs that always evaluate to false.

**What you expected to happen**:
Extensions to pass and handle properly IgnoreOperationAnnotation. 

PS. IMO the whole option handling is too hard to follow.

- Gardener version:
- Kubernetes version (use `kubectl version`):
- Cloud provider or hardware configuration:
- Others:

**Comment by timuthy** [comment]
> PS. IMO the whole option handling is too hard to follow.

+1000

**Comment by rfranzke** [comment]
➕💯0️⃣ as well :)
